src/nevermined/Nfts.ts

```diff
@@ -512,4 +515,140 @@ export class Nfts extends Instantiable {
+                    description: ''
+                },
+                serviceAgreementTemplate: nftSalesServiceAgreementTemplate
+            }
+        }
+
+        const saveResult = await this.nevermined.metadata.storeService(
+            agreementId,
+            nftSalesServiceAgreement
+        )
+
+        if (saveResult) {
+            return agreementId
+        } else {
+            throw Error(`Error saving ${agreementId} to MetadataDB`)
+        }
+    }
+
+    /**
+     * Buys a number of listed goods on secondary markets.
+     * @param consumer The account of the buyer/consumer.
+     * @param seller The account of the seller.
+     * @param nftAmount The number of assets to buy. 1 by default.
+     * @param ddo The DDO of the NFT.
+     * @param provider The provider address.
+     * @param agreementId The agreementId of the initial sales agreement created off-chain.
+     * @returns true if the buy was successful.
+     */
+    public async buySecondaryMarketNft(
+        consumer: Account,
+        seller: Account,
+        nftAmount: number = 1,
+        ddo: DDO,
+        provider: Account,
+        agreementId: string
+    ): Promise<boolean> {
+        const { nftSalesTemplate } = this.nevermined.keeper.templates
+        const service = await this.nevermined.metadata.retrieveService(agreementId)
+        const assetRewards = getAssetRewardsFromService(service)
+
+        const result = await nftSalesTemplate.createAgreementFromDDO(
+            agreementId,
+            ddo,
+            assetRewards,
+            consumer,
+            nftAmount,
+            provider || seller,
+            consumer,
+            service as TxParameters
+        )
+
+        if (!result) throw new Error('Creating buy agreement failed')
+
+        const payment = findServiceConditionByName(service, 'lockPayment')
+
+        let receipt = await this.nevermined.agreements.conditions.lockPayment(
+            agreementId,
+            ddo.id,
+            assetRewards.getAmounts(),
+            assetRewards.getReceivers(),
+            payment.parameters.find(p => p.name === '_tokenAddress').value as string,
+            consumer
+        )
+
+        if (!receipt) throw new Error('LockPayment Failed.')
+
+        receipt = await this.nevermined.agreements.conditions.transferNft(
+            agreementId,
+            ddo,
+            assetRewards.getAmounts(),
+            assetRewards.getReceivers(),
+            nftAmount,
+            seller,
+            assetRewards as TxParameters
+        )
+
+        if (!receipt) throw new Error('TranferNft Failed.')
+
+        receipt = await this.nevermined.agreements.conditions.releaseNftReward(
+            agreementId,
+            ddo,
+            assetRewards.getAmounts(),
+            assetRewards.getReceivers(),
+            nftAmount,
+            seller
+        )
+
+        if (!receipt) throw new Error('ReleaseBftReward Failed.')
+        return true
+    }
```

Comment:
again this shouldn't work because we don't have the seller account

Reply:
I'm wondering what will be the solution here. For now it seems there are two options: make it work like this(harder) or split into a 3 step process(easier). ❓